### PR TITLE
Reusable menu component, todo page

### PR DESCRIPTION
**Pull request by barrystaes**
While looking into (the awesome) react-starter i fixed/moved some trivial things.
- [x] Added React components: MainMenu (shown above every page) and TodolistMenu.
- [x] Added TodoPage to show TodolistMenu, and use with
- [x] TodoPage subroutes.

In hindsight i should have done this in two seperate PR.. (components and todopage) 
Let me know what you think.

_Note: A argument against accepting this PR might be that the things i did here (coincidentally?) are excellent exercises anyone might like to go through. They gave me some insights on how this react-starter kit works._

### CODE REVIEW

**Comment by barrystaes** [comment]
I'd like to put the "update data" button that triggers `Application.update()` on the TodoList page, but i dont know how to correctly refer to the `Application` instance.

File: app/mainRoutes.jsx [hunk 1]
```diff
@@ -12,9 +12,11 @@ module.exports = (
 	<Route name="app" path="/" handler={require("./Application")}>
 		<Route name="some-page" path="/some-page" handler={require("react-proxy!./SomePage")} />
 		<Route name="readme" path="/readme" handler={require("react-proxy!./Readme")} />
-		<Route name="todolist" path="/:list" handler={require("./TodoList")} />
-		<Route name="todoitem" path="/todo/:item" handler={require("./TodoItem")} />
+		<Route name="todo" path="/todo" handler={require("react-proxy!./TodoPage")} >
```
Comment (sokra) [inline app/mainRoutes.jsx]: Try to no use `react-proxy` here. This shouldn't be loaded on demand. (On demand loaded stuff cannot be prerendered.)

File: app/TodoPage/TodoList/index.jsx [hunk 2]
```diff
@@ -32,8 +32,7 @@ var TodoList = React.createClass({
 		var items = this.state.items;
 		var info = this.state.info;
 		return <div>
-			<h2>Todolist</h2>
-			<Link to="home">Home</Link>
+			<h3>Todolist: {id}</h3>
```
Comment (sokra) [inline app/TodoPage/TodoList/index.jsx]: Why removing the `{id}`?

Comment (sokra) [inline app/TodoPage/TodoList/index.jsx]: Ah sorry. you added it... That's fine.

File: app/Application/MainMenu.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+var React = require("react");
+var Link = require("react-router").Link;
+
+var MainMenu = React.createClass({
+    render: function() {
+        return <div>
+            <h2>MainMenu:</h2>
```
Comment (sokra) [inline app/Application/MainMenu.jsx]: use tabs...

**Comment by sokra** [comment]
> I'd like to put the "update data" button that triggers Application.update() on the TodoList page, but i dont know how to correctly refer to the Application instance.

For the `update()` use an Action, which triggers the update and is bound in `mainStores` like the other actions.

**Comment by sokra** [comment]
Normally components shouldn't access the stores except for reading in `getState()`. Everything else should be done with actions. (I've done it wrong in the `update()` method)

**Comment by barrystaes** [comment]
Thanks @sokra for the insights! 
I'll look into the update action, its new to me..

**Comment by barrystaes** [comment]
I looked at actions, but dont grasp how to implement what you're suggesting. Could you lead the way?

I get Flux, but haven't looked at implementations much. Just found out `react-starter` uses `items-store`. On a related Flux note, @sokra how do you feel about using Marty.js (or Reflux.js) with react-starter? The argument for using items store would be welcome in my HowStuffWorks writeup.

**Comment by sokra** [comment]
I didn't want to write a new module. I actually tried using Reflux.js, but couldn't find a good workflow for prerendering, optimistic updates and merging of multiple writes. You could do it manually but that is something I would expect from a flux implementation. items-store is a very simple store base class that coordinate this behavior (the repo actually also contains a simple helper for actions and a react mixin, but these are theoretically independant and you don't have to use them with the store).

items-store allows to serialize and deserialize the store data to inject the data after prerendering. It manages optimistic updates and merges multiple writes. But items-store only offers a simple key-value store API and forces you to map more complex operations (via store-helpers) to this model. It's just a caching layer with events to the remote API (which need to be provided in the constructor).

items-store basically provides a "raw data" store, while other implementions give you the ablility to write higher level stores. In items-store you need to put this higher-level behavior in store-helpers. The advantage is that this way basic functionality is already provided by items-store.

---

Now there is an alternative to items-store, which could provide "prerendering" part too: http://fluxible.io/ they use dehyrade and rehyrade to (de)serialize to data from the stores and provide it to the client stores.

**Comment by barrystaes** [comment]
Thanks for the store insights. Could you amend the `How the stores work` section?

**Comment by sokra** [comment]
Thanks

**Pull request merged**